Contributions/Kernel_Self_Monitoring.md

```diff
@@ -0,0 +1,117 @@
+# **Kernel Self Monitoring**
+
+## Index
+[References](#References)
+
+[Purpose of the Document](#Purpose-of-the-Document)
+
+[Structure of the Document](#Structure-of-the-Document)
+
+[Introduction - Linux Kernel: practical safety problems](#Introduction---Linux-Kernel-practical-safety-problems)
+
+[The concept - Safety-oriented self monitoring](#The-concept---Safety-oriented-self-monitoring)
+
+[Constraints for a safety-oriented kernel self monitor](#Constraints-for-a-safety-oriented-kernel-self-monitor)
+
+[Targets for self monitoring activity](#Targets-for-self-monitoring-activity)
+
+
+## **References**
+
+## **Purpose of the Document**
+These are guidelines for safety-hardening certain aspects of the Linux kernel, through self monitoring.
+
+They can act as a touchstone against candidate designs, to avoid known pitfalls. 
+
+## **Structure of the document**
+- The document presents a brief overview of what safety problems can be found while using Linux, and how they can be mitigated through self-monitoring.
+- The next sections, instead, discuss more in detail typical pitfalls that must be avoided, to make the monitoring useful and what goals should be targeted.
+
+## **Introduction - Linux Kernel: practical safety problems**
+The Linux Kernel has many desirable features that make it a prime candidate for many use cases.
+
+However, its lacks of both an intrinsic safety design and safety-oriented features make it challenging to use it as-it-is, for safety applications.
+
+Therefore, there is a risk of having a Linux-based system to experience undetected internal interference, which might compromise the set of safety goals the system must support.
+
+It is a customary mitigation to introduce system-level monitoring, in various forms.
+
+To mention a couple of examples:
+- Watchdogs, to detect abnormal delays either in processing events or performing actions.
+- Output vetting, to detect abnormal actuator control signals produced by the system.
+
+These external safety measures, though, are not always sufficient, because they might have limited ability to observe the internal status of the kernel, or it might be desirable to exert a tighter control over the evolution of its internal states.
+This is where the concept of kernel self monitoring comes into the picture.
+
+
+## **The concept - Safety-oriented self monitoring**
```

Comment:
While I agree on the technical content, I think that in this doc we are missing the theoretical principles, that BTW are quite simple:
1) The Self Monitoring must be developed or qualified with a systematic capability level that is equal or higher than the ASIL or SIL level associated with the safety claim that it supports
2) The scope of the monitored part of the Kernel and the scope of the self monitoring part shall be clearly identified, including part of the design or code that can be common between the two 
3) A safety analysis on the monitored code shall define the dangerous failure modes originating from it that would violate the target safety claim
4) Assuming the lack of interference from the monitored code and from any other code running in the Kernel, the safety monitor shall be verified to be effective against a subset of the dangerous failure modes mentioned in 3). Any residual dangerous failure mode shall be covered by additional mitigations (that are out of the scope of this document)
5) A comprehensive analyses on the possible interference failure modes originating from the monitored code or any other code running in the Kernel shall be carried over. With respect to such analysis the dangerous interference failure modes (leading to a violation of the target safety claim) shall be identified and mitigated through adequate measures.
For example adequate measures could range from Assumption on Use or proper configurations that would lead to the avoidance of such failure modes, designing and implementing additional detection mechanism or the qualification of the code generating interference up to a level of systematic capability that is equal or higher than ASIL or SIL allocated to the target safety claim.

**Note**: When designing the safety monitors and any additional mitigation measure, also availability, security and performance requirements shall be considered.
E.g. a measure making the overall safety function unavailable is perfectly safe but it would be useless